packages/dapp/src/hooks/useKeysManager.ts

```diff
@@ -0,0 +1,156 @@
+import type { VerificationMethodReference } from '@windingtree/org.json-schema/types/org.json';
+import type { GenericStateRecord } from '../store';
+import { useState, useCallback } from 'react';
+import { useAppReducer } from '../store/reducer';
+
+export const allowedKeysTypes: string[] = ['EcdsaSecp256k1VerificationKey2019'];
+
+export type KeyType =
+	typeof allowedKeysTypes[number];
+
+export enum RevocationReason {
+  "unspecified",
+  "keyCompromise",
+  "affiliationChanged",
+  "superseded",
+  "cessationOfOperation",
+  "certificateHold",
+  "privilegeWithdrawn"
+}
+
+export interface KeyRecord extends GenericStateRecord {
+	id: string; // Unique Id generated using `simpleUid()` 
+							// from @windingtree/org.id-utils package. 
+							// For example see the docs: 
+							// https://windingtree.github.io/org.id-sdk/#/build/utils/index?id=simpleuidlength
+	type: KeyType;
+	publicKey: string; // Ethereum account address
+	tag: string; // Unique across records formatted string
+	note?: string; // Key description
+	revocation?: VerificationMethodReference['verificationMethodRevocation']; // see this type for details
+}
+
+export type Keys = KeyRecord[];
+
+export type UseKeysManager = [
+  addKey: Function,
+  updateKey: Function,
+  removeKey: Function,
+  revokeKey: Function,
+  loading: boolean,
+  error: string | undefined,
+];
+
+// UseKeysManager react hook
+export const useKeysManager = (): UseKeysManager => {
+  const [state, dispatch] = useAppReducer();
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | undefined>(undefined);
+
+  const findKeyById = (id:string) => (state.keys as Keys).find((key) => key.id == id)
+  const findKeyByTag = (tag:string) => (state.keys as Keys).find((key) => key.tag == tag)
+
+  const addKey = useCallback((record:KeyRecord) => {
```

Comment:
`record` must be validated against the schema. 
Typescript type is not enough, it is for compilation time chack. But you must validate records in run time
So, create a json schema for this type and validate a record with it.
Here is the example: https://windingtree.github.io/org.id-sdk/#/build/utils/index?id=validating-an-object-against-the-schema

Reply:
:heavy_check_mark:

Reply:
✔️